libs/langchain/tests/unit_tests/retrievers/document_compressors/test_encoded_chain_extract.py

```diff
@@ -0,0 +1,45 @@
+"""Unit tests for LLMEncodedChainExtractor."""
+import pytest
+
+from langchain.retrievers.document_compressors.encoded_chain_extract import (
+    SequenceListParser,
+    extract_numbered_sequences,
+    number_sequences,
+)
+from langchain.schema.output_parser import OutputParserException
+
+
+def test_parser() -> None:
+    parser = SequenceListParser()
+    assert parser.parse("1,2,3") == [1, 2, 3]
+    assert parser.parse("1-3") == [1, 2, 3]
+    assert parser.parse("1-3,5") == [1, 2, 3, 5]
+    assert parser.parse("1-3,5-7") == [1, 2, 3, 5, 6, 7]
+    with pytest.raises(OutputParserException):
+        assert parser.parse("1,6,10-8")
+    with pytest.raises(OutputParserException):
+        assert parser.parse("1,6,a")
+
+
+@pytest.mark.requires("spacy")
+def test_number_sequences() -> None:
```

Comment:
is this deterministic or using  model under the hood?

either way i think needing to install en_core_web_sm for unit tests may be a bit heavy, should we move it to integration testing?

Reply:
This is done.